gulp/sass.js

```diff
@@ -33,26 +37,78 @@ gulp.task('sass', function () {
+    .on('end', showDeprecationError(deprecationWarningList))
 })
 
 gulp.task('sass-documentation', function () {
+  const deprecationWarningList = {}
   return gulp.src(config.paths.docsAssets + '/sass/*.scss', { sourcemaps: true })
-    .pipe(sass.sync({ outputStyle: 'expanded', logger: sass.compiler.Logger.silent  }).on('error', sass.logError))
+    .pipe(sass.sync({
+      outputStyle: 'expanded',
+      logger: customLogger(deprecationWarningList)
+    }).on('error', sass.logError))
     .pipe(gulp.dest(config.paths.public + '/stylesheets/', { sourcemaps: true }))
+    .on('end', showDeprecationError(deprecationWarningList))
 })
 
 // Backward compatibility with Elements
 
 gulp.task('sass-v6', function () {
+  const deprecationWarningList = {}
   return gulp.src(config.paths.v6Assets + '/sass/*.scss', { sourcemaps: true })
     .pipe(sass.sync({
       outputStyle: 'expanded',
-      logger: sass.compiler.Logger.silent,
+      logger: customLogger(deprecationWarningList),
       includePaths: [
         'node_modules/govuk_frontend_toolkit/stylesheets',
         'node_modules/govuk-elements-sass/public/sass',
         'node_modules/govuk_template_jinja/assets/stylesheets'
       ]
     }).on('error', sass.logError))
     .pipe(gulp.dest(config.paths.public + '/v6/stylesheets/', { sourcemaps: true }))
+    .on('end', showDeprecationError(deprecationWarningList))
 })
+
+// Custom logs to support the Dart Sass transition
+
+function customLogger (warningList) {
+  return {
+    debug: (debug, meta) => {
+      console.log(debug)
+      console.log(meta.stack)
+    },
+    warn: (warning, meta) => {
+      if (warning.match(/^[0-9]+ repetitive deprecation warnings omitted\.$/)) {
+        return
+      }
+      if (meta.deprecation === true && warning.includes('More info and automated migrator: https://sass-lang.com/d/slash-div')) {
+        const match = meta.stack.match(/node_modules[/\\]([^/\\]+)/)
+        const project = match ? match[1] : 'GOVUK Prototype Kit'
+        warningList[project] = warningList[project] || 0
+        warningList[project] += 1
+      } else {
+        console.warn(warning, meta.stack)
+      }
+    }
+  }
+}
+
+function showDeprecationError (warningList) {
+  return function () {
+    if (Object.keys(warningList).length === 0) {
+      return;
+    }
+    const fmtStart = '\x1b[33m'
+    const fmtEnd = '\x1b[0m'
+    const em = '\x1b[4m'
+    const fmtRestart = `${fmtEnd}${fmtStart}`
+    Object.keys(warningList).forEach(project => console.warn(
+      `${fmtStart}${em}${project}${fmtRestart} uses a forward slash for division ${em}${warningList[project]} time${warningList[project] ? 's' : ''}${fmtRestart}.${fmtEnd}`))
+
+    console.warn([
+      fmtStart,
+      'This is deprecated as documented at https://sass-lang.com/d/slash-div.',
+      'This won\'t cause any problems with your prototype but it would we would encourage them to move to Dart Sass.',
```

Comment:
What if the deprecated code is being used in the user's own Sass?

Reply:
I'm not sure how helpful this is for users of the Prototype Kit? I'm wondering what we'd expect them to do with this information.

Reply:
@lfdebrux if it's in their sass it will list the file names.

Reply:
@vanitabarrett I think this is more useful for other design systems - e.g. if `hmrc-frontend` introduced a libsass style division they might not realise if the warnings are fully suppressed.

Reply:
@natcarey will it still say 'we would encourage them to move to Dart Sass'?